src/canister_tests/src/tests.rs

```diff
@@ -52,3 +59,23 @@ fn ii_canister_can_be_upgraded_and_rolled_back() {
     framework::upgrade_ii_canister(&env, canister_id, framework::II_WASM_PREVIOUS.clone());
     api::health_check(&env, canister_id);
 }
+
+#[test]
+fn registration_with_mismatched_sender_fails() {
+    let env = StateMachine::new();
+    let canister_id = framework::install_ii_canister(&env, framework::II_WASM_PREVIOUS.clone());
+    let challenge = api::create_challenge(&env, canister_id);
+    let result = register(
+        &env,
+        canister_id,
+        other_principal(),
+        framework::some_device_data(),
```

Comment:
Agreed. But I'd rather do numbered sets (i.e. `pubkey_1`, `device_data_1`, `principal_1`) because we also need the different devices to test device registration, etc. And it does not make a lot of sense for alice to add bobs device to her account.